services/wallet/api.go

```diff
@@ -635,3 +636,15 @@ func (api *API) FetchChainIDForURL(ctx context.Context, rpcURL string) (*big.Int
 	client := ethclient.NewClient(rpcClient)
 	return client.ChainID(ctx)
 }
+
+func (api *API) WCPairSessionProposal(ctx context.Context, sessionProposalJSON string) (*wc.PairSessionResponse, error) {
```

Comment:
hmm i've never seen us pass arguments from the client explicitly as a JSON string, any reason why we do the unmarshaling manually here?

Reply:
They come as `JSON` from the `WebView` through the `WalletConnect SDK`. It seemed pointless to parse it in Nim to forward them to status-go. Is there any value in having the parsing and logic in Nim?

Reply:
Ohh alright, have yet to check the status-desktop one. I guess it makes sense if it's data we're passing around and not building ourselves